src/hardware/disk_noise.cpp

```diff
@@ -0,0 +1,479 @@
+		          candidate.c_str());
+		return;
+	}
+
+	LOG_WARNING("DISKNOISE: Failed to find FLAC sample: %s", path.c_str());
+}
+
+void DiskNoiseDevice::LoadSeekSamples(const std::vector<std::string>& paths)
+{
+	for (const auto& path : paths) {
+		if (path.empty()) {
+			// Placeholder for missing sample files
+			seek_samples_.emplace_back();
+			continue;
+		}
+
+		std::vector<float> sample;
+		LoadSample(path, sample);
+		if (!sample.empty()) {
+			seek_samples_.push_back(std::move(sample));
+		} else {
+			seek_samples_.emplace_back();
+		}
+	}
+
+	// Assign weights with descending priority (most weight to first samples)
+	const int max_weight = 10;
+	seek_sample_weights_.resize(seek_samples_.size());
+	for (unsigned int i = 0; i < seek_samples_.size(); ++i) {
+		seek_sample_weights_[i] = static_cast<int>(max_weight - i);
+	}
+}
+
+int DiskNoiseDevice::ChooseWeightedSeekIndex() const
+{
+	int total_weight = 0;
+	for (unsigned int i = 0; i < seek_samples_.size(); ++i) {
+		if (!seek_samples_[i].empty()) {
+			total_weight += seek_sample_weights_[i];
+		}
+	}
+
+	if (total_weight == 0) {
+		return 0;
+	}
+
+	const int r = static_cast<int>(rand()) % total_weight;
+	int sum     = 0;
+	for (unsigned int i = 0; i < seek_samples_.size(); ++i) {
+		if (seek_samples_[i].empty()) {
+			continue;
+		}
+
+		sum += seek_sample_weights_[i];
+		if (r < sum) {
+			return i;
+		}
+	}
+	return 0;
+}
+DiskNoiseDevice::DiskNoiseDevice(const bool enable_disk_noise,
+                                 const std::string& spin_up_sample_path,
+                                 const std::string& spin_sample_path,
+                                 const std::vector<std::string>& seek_sample_paths,
+                                 const float& spin_volume,
+                                 const float& seek_volume, bool loop_spin_sample)
+        : spin_volume_(spin_volume),
+          seek_volume_(seek_volume),
+          spin_up_sample_(),
+          spin_sample_(),
+          current_seek_sample_(),
+          seek_samples_(),
+          seek_sample_weights_()
+{
+	enable_disk_noise_ = enable_disk_noise;
+	loop_spin_sample_  = loop_spin_sample;
+
+	if (!enable_disk_noise_) {
+		LOG_INFO("DISKNOISE: Disk noise emulation disabled");
+		return;
+	}
+
+	LoadSample(spin_up_sample_path, spin_up_sample_);
+	LoadSample(spin_sample_path, spin_sample_);
+	LoadSeekSamples(seek_sample_paths);
+
+	std::lock_guard<std::mutex> lock(device_mutex_);
+
+	// Lazily create the static mixer channel if it doesn't exist yet
+	if (!mix_channel_) {
+		MIXER_LockMixerThread();
+		mix_channel_ = MIXER_AddChannel(AudioCallback,
+		                                SampleRate,
+		                                ChannelName::DiskNoise,
+		                                {ChannelFeature::Stereo});
+		mix_channel_->Enable(true);
+		MIXER_UnlockMixerThread();
+		float vol_gain = percentage_to_gain(100);
+		mix_channel_->SetAppVolume({vol_gain, vol_gain});
+	}
+
+	active_devices_.push_back(this);
```

Comment:
You won't need the extra mutex if you introduce `DiskNoises` as suggested. That will then create/destroy the channel only once.